### ISSUE TITLE
jax.scipy.stats.cauchy.ppf/isf: no quantile-domain handling — ppf(0) returns large finite value (wrong sign in float32), q outside [0,1] returns garbage

### ISSUE THREAD

**Issue by OE-GOD** [jax-ml/jax#38318]
### Description

`jax.scipy.stats.cauchy.ppf` / `isf` evaluate the raw inverse-CDF formula with no quantile-domain handling:

```python
import jax
import jax.scipy.stats
import scipy.stats
jax.config.update('jax_enable_x64', True)

print(jax.scipy.stats.cauchy.ppf(0.0))   # -1.633123935319537e+16  (scipy: -inf)
print(jax.scipy.stats.cauchy.ppf(2.0))   # 5443746451065123.0      (scipy: nan)
```

In float32 it's worse — the result at the boundary has the **wrong sign**, because the float32 rounding of `pi/2` lands on the other side of the tan pole:

```python
print(jax.scipy.stats.cauchy.ppf(jax.numpy.float32(0.0)))
# +22877296.0 — the 0th percentile reported as a large positive value
```

### Root cause

`jax/_src/scipy/stats/cauchy.py` computes `lax.tan(lax.sub(lax.mul(pi, q), half_pi))` directly with no boundary or domain logic. scipy's convention (`ppf(0)`/`ppf(1)` → support endpoints, `q ∉ [0,1]` → `nan`) is already implemented in JAX itself by `uniform.ppf`:

```python
return jnp.where(jnp.isnan(q) | (q < 0) | (q > 1), np.nan, ...)
```

so this is also an internal inconsistency between distributions in `jax.scipy.stats`. The same issue affects `cauchy.isf`, and `gumbel_l.ppf` / `gumbel_r.ppf` return `nan` at `q=0`/`q=1` instead of the correct infinite limits.

**Suggested fix:** clamp boundaries and mask the domain as `uniform.ppf` does, e.g.

```python
out = jnp.where(q == 0, -jnp.inf, jnp.where(q == 1, jnp.inf, formula))
return jnp.where((q < 0) | (q > 1) | jnp.isnan(q), jnp.nan, out)
```

Reproduces on 0.10.1 and current `main`.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.10.1
jaxlib: 0.10.1
numpy:  2.4.6
python: 3.12.13
device: CPU (arm64, macOS)
```

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
